completion/src/lib.rs

```diff
@@ -204,10 +202,16 @@ impl<E: TypeEnv> OnFound for Suggest<E> {
     }
 }
 
+enum Location<'a> {
```

Comment:
Perhaps `MatchState` would be a better name (a bit bland but at least it is explicit).

Reply:
I like that!